packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -443,29 +466,21 @@ async function notMemoizedbase64({ file, args = {}, reporter }) {
       pipeline
     )
   }
-
   const [buffer, info] = await pipeline.toBufferAsync()
-  const originalName = file.base
-
+  const base64output = buffer.toString(`base64`)
+  if (cache) {
+    await cache.set(base64CacheKey(file.internal.contentDigest, options), { base64: base64output, info })
+  }
   return {
-    src: `data:image/${info.format};base64,${buffer.toString(`base64`)}`,
+    src: `data:image/${info.format};base64,${base64output}`,
```

Comment:
Let's create this object first and _then_ cache it. May as well, right? Then we save ourselves from having to create this twice.